src/main.cpp

```diff
@@ -94,6 +94,7 @@ struct COrphanTx {
 };
 map<uint256, COrphanTx> mapOrphanTransactions GUARDED_BY(cs_main);
 map<uint256, set<uint256> > mapOrphanTransactionsByPrev GUARDED_BY(cs_main);
```

Comment:
It would be more efficient to store map<uint256, COrphanTx>::iterators instead of uint256's